models/balancer/arbitrum/balancer_v2_arbitrum_pools_fees.sql

```diff
@@ -9,36 +10,38 @@
         post_hook='{{ expose_spells(\'["arbitrum"]\',
                                     "project",
                                     "balancer_v2",
-                                    \'["metacrypto", "jacektrocinski"]\') }}'
+                                    \'["metacrypto", "jacektrocinski", "thetroyharris"]\') }}'
     ) 
 }}
 
 {% set event_signature = '0xa9ba3ffe0b6c366b81232caab38605a0699ad5398d6cce76f91ee809e322dafc' %}
 {% set project_start_date = '2021-08-26' %}
 
 WITH registered_pools AS (
-    SELECT DISTINCT
-        `poolAddress` AS pool_address
+    SELECT
+        DISTINCT poolAddress AS pool_address
     FROM
         {{ source ('balancer_v2_arbitrum', 'Vault_evt_PoolRegistered') }}
+    {% if is_incremental() %}
+    WHERE evt_block_time >= DATE_TRUNC('day', NOW() - interval '7' day)
+    {% endif %}
 )
 SELECT
+    'arbitrum' AS blockchain,
     logs.contract_address,
     logs.tx_hash,
     logs.tx_index,
     logs.index,
     logs.block_time,
     logs.block_number,
-    bytea2numeric_v3 (SUBSTRING(logs.data FROM 32 FOR 64)) * 1 AS swap_fee_percentage
+    bytea2numeric(bytearray_ltrim(logs.data)) AS swap_fee_percentage
 FROM
     {{ source ('arbitrum', 'logs') }}
     INNER JOIN registered_pools ON registered_pools.pool_address = logs.contract_address
-WHERE logs.topic1 = '{{ event_signature }}'
+WHERE CAST(logs.topic0 AS VARCHAR) = '{{ event_signature }}'
     {% if not is_incremental() %}
-    AND logs.block_time >= '{{ project_start_date }}'
+    AND logs.block_time >= CAST('{{ project_start_date }}' AS TIMESTAMP)
```

Comment:
the preferred approach here:
`TIMESTAMP '{{ project_start_date }}` rather than casting

rule of thumb:
- if using a string literal value hardcoded, use datatype in front (i.e. `UINT256 '0'` or `TIMESTAMP '<date>'`
- if a column, `CAST()` is preferred

Reply:
Appreciate you letting me in on the rule of thumb, I will remember that.